python-checks/src/main/java/org/sonar/python/checks/regex/SingleCharCharacterClassCheck.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.regex;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonarsource.analyzer.commons.regex.RegexParseResult;
+import org.sonarsource.analyzer.commons.regex.finders.SingleCharCharacterClassFinder;
+
+@Rule(key = "S6397")
+public class SingleCharCharacterClassCheck extends AbstractRegexCheck {
+
+  @Override
+  public void checkRegex(RegexParseResult regexParseResult, CallExpression regexFunctionCall) {
+    Optional.ofNullable(regexFunctionCall.calleeSymbol())
+      .flatMap(symbol -> Optional.ofNullable(symbol.fullyQualifiedName()))
+      .filter(fqn -> lookedUpFunctions().containsKey(fqn))
+      .filter(fqn -> !getFlagSet(regexFunctionCall, fqn).contains(Pattern.COMMENTS))
```

Comment:
```suggestion
      .filter(fqn -> !regexParseResult.getResult().activeFlags().contains(Pattern.COMMENTS))
```
Would be an alternative to not expose the getFlagSet function.

Reply:
Indeed, thanks!